distribution/pull_v2.go

```diff
@@ -665,14 +665,20 @@ func (p *v2Puller) pullSchema2Layers(ctx context.Context, target distribution.De
 		}
 	}
 
+	// Assume that the operating system is the host OS if blank, and validate it
+	// to ensure we don't cause a panic by an invalid index into the layerstores.
+	if layerStoreOS != "" && !system.IsOSSupported(layerStoreOS) {
+		return "", system.ErrNotSupportedOperatingSystem
```

Comment:
Perhaps we could wrap this error to make it more user-friendly (I kept the existing error for now, and there's a bit of a mix how we use it (some places wrap it, other places don't)